api/v1alpha1/warehouse_types.go

```diff
@@ -267,6 +284,127 @@ type WarehouseStatus struct {
 	LastFreight *FreightReference `json:"lastFreight,omitempty" protobuf:"bytes,5,opt,name=lastFreight"`
+	// DiscoveredArtifacts holds the artifacts discovered by the Warehouse.
+	DiscoveredArtifacts *DiscoveredArtifacts `json:"discoveredArtifacts,omitempty" protobuf:"bytes,7,opt,name=discoveredArtifacts"`
+}
+
+// DiscoveredArtifacts holds the artifacts discovered by the Warehouse.
+type DiscoveredArtifacts struct {
+	// Commits holds the Git commits discovered by the Warehouse.
+	Commits []GitDiscoveryResult `json:"commits,omitempty" protobuf:"bytes,1,rep,name=commits"`
+
+	// Images holds the images discovered by the Warehouse.
+	Images []ImageDiscoveryResult `json:"images,omitempty" protobuf:"bytes,2,rep,name=images"`
+
+	// Charts holds the charts discovered by the Warehouse.
+	Charts []ChartDiscoveryResult `json:"charts,omitempty" protobuf:"bytes,3,rep,name=charts"`
+}
+
+// GitDiscoveryResult represents the result of a Git discovery operation for a
+// GitSubscription.
+type GitDiscoveryResult struct {
+	// RepoURL is the repository URL of the GitSubscription.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:Pattern=`^https?://(\w+([\.-]\w+)*@)?\w+([\.-]\w+)*(:[\d]+)?(/.*)?$`
+	RepoURL string `json:"repoURL" protobuf:"bytes,1,opt,name=repoURL"`
+
+	// Commits is a list of commits discovered by the Warehouse for the
+	// GitSubscription. An empty list indicates that the discovery operation was
+	// successful, but no commits matching the GitSubscription criteria were found.
+	// +optional
+	Commits []DiscoveredCommit `json:"commits,omitempy" protobuf:"bytes,2,rep,name=commits"`
+}
+
+// DiscoveredCommit represents a commit discovered by a Warehouse for a
+// GitSubscription.
+type DiscoveredCommit struct {
+	// ID is the identifier of the commit. This typically is a SHA-1 hash.
+	// +kubebuilder:validation:MinLength=1
+	ID string `json:"id,omitempty" protobuf:"bytes,1,opt,name=id"`
+
+	// Branch is the branch in which the commit was found. This field is
+	// optional, and populated based on the CommitSelectionStrategy of the
+	// GitSubscription.
+	// +optional
+	Branch string `json:"branch,omitempty" protobuf:"bytes,2,opt,name=branch"`
+
+	// Tag is the tag that resolved to this commit. This field is optional, and
+	// populated based on the CommitSelectionStrategy of the GitSubscription.
+	// +optional
+	Tag string `json:"tag,omitempty" protobuf:"bytes,3,opt,name=tag"`
+
+	// Subject is the subject of the commit (i.e. the first line of the commit
+	// message).
+	// +optional
+	Subject string `json:"subject,omitempty" protobuf:"bytes,4,opt,name=subject"`
+
+	// Author is the author of the commit.
+	// +optional
+	Author string `json:"author,omitempty" protobuf:"bytes,5,opt,name=author"`
+
+	// CreatedAt is the time at which the commit or annotated tag was created.
+	// +optional
+	CreatedAt *metav1.Time `json:"createdAt,omitempty" protobuf:"bytes,6,opt,name=createdAt"`
```

Comment:
In Git concepts, this would be named a `CreatorDate`. Based on my observations, Kargo does however lean towards not doing 1:1 copying of these type of terms, and because of this I opted for the more Kubernetes-like `CreatedAt`.

Reply:
I'm fine with either.